### PR TITLE
#16775 remove exception when no user specified

### LINKED ISSUE #16775: SSH tunnel requires username
#### System information: 
- Operating system (distribution) and version
- MacOS 11.6.6
- DBeaver version
- Version 22.1.0.202206051253
- Additional extensions

#### Connection specification:
- Database name and version
- MariaDB 10.3.34
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 
SSH tunnel
#### Describe the problem you're observing:
Unable to set up a connection to SQL server via SSH tunnel. The Config screen requires a user name and password. This should not be required with authentication mode set to Public Key, and private key selected 
#### Include any warning/errors/backtraces from the logs

2022-06-10 14:40:50.291 - org.jkiss.dbeaver.model.exec.DBCException: Can't initialize tunnel
org.jkiss.dbeaver.model.exec.DBCException: Can't initialize tunnel
        at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:913)
        at org.jkiss.dbeaver.runtime.jobs.ConnectJob.run(ConnectJob.java:70)
        at org.jkiss.dbeaver.runtime.jobs.ConnectionTestJob.run(ConnectionTestJob.java:103)
        at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.jkiss.dbeaver.DBException: SSH user not specified
        at org.jkiss.dbeaver.model.net.ssh.SSHImplementationAbstract.loadConfiguration(SSHImplementationAbstract.java:203)
        at org.jkiss.dbeaver.model.net.ssh.SSHImplementationAbstract.initTunnel(SSHImplementationAbstract.java:98)
        at org.jkiss.dbeaver.model.net.ssh.SSHTunnelImpl.initializeHandler(SSHTunnelImpl.java:76)
        at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:908)
        ... 4 more
2022-06-10 14:40:50.345 - SSH user not specified

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes #16775 
Release: https://github.com/dbeaver/dbeaver/pull/17706

### CODE REVIEW

**Comment by serge-rider** [comment]
User name is required in general. It may be not required if we use public key based auth only.

File: plugins/org.jkiss.dbeaver.net.ssh/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationAbstract.java [hunk 1]
```diff
@@ -199,11 +199,10 @@ private static SSHHostConfiguration loadConfiguration(@NotNull DBWHandlerConfigu
         if (port == 0) {
             throw new DBException("SSH port not specified");
         }
-        if (CommonUtils.isEmpty(username)) {
+        final SSHAuthConfiguration authentication;
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.net.ssh/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationAbstract.java]: Is this variable used somewhere?

**Comment by Matvey16** [comment]
Clicking test tunnel now asks for passphrase even when public key is selected

dbeaver_tDm7vD2XBW

**Comment by Matvey16** [comment]
Verified

**Pull request merged**